### PR TITLE
eHealth Exchange Demonstration Using CONNECT 5.4

### PR DESCRIPTION
**Pull request by ddecouteau**
This creates a CES library leap-ces-ccda-orchestration-embedded for consumption by CONNECT 5.4 which requires JDK 1.7 build, and Testing using SOAP-UI.  The README  within this project for Build and Testing is in-progress.

### CODE REVIEW

**Comment by ddecouteau** [comment]
Build is failing on CONNECT core and commontypes libs which are provided in CONNECT deployment

**Comment by ddecouteau** [comment]
@sgroh can you exclude leap-ces-ccda-orchestration-embedded from primary github action and create an action to create lib built on JDK 1.7?

File: leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/model/card/Card.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
```
Comment (mojitoj) [inline leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/model/card/Card.java:2]: Don't we have this model POJOs in the java-client? Is there a reason we don't want to use that dependency?

File: leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/orchestration/service/CCDAOrchestrationService.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    private PatientConsentConsultXacmlRequestWithData xacmlRequestWithData;
+    private XacmlRequest xacmlRequest;
+
+    private String decision;
+    private String action = "";
+    private String label = "";
+
+
+    //@Value("${cds.host.url}")
+    private String CDS_HOST = "http://sdhc-leap.appspot.com";
```
Comment (mojitoj) [inline leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/orchestration/service/CCDAOrchestrationService.java]: should these be hard coded?

**Review by mojitoj — COMMENTED** [review]
just left a couple of minor comments.

File: leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/model/card/Card.java (hunk 1 above)
Comment (ddecouteau) [reply leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/model/card/Card.java:2]: Java version issue was being thrown at runtime when our package was referenced.  CONNECT 5.4 is built to JDK 1.7.  By going down this path, which is redundant, a single lib build to 1.7, containing all necessary is created.   Is redundant, but works for demonstration purpose at this time.

File: leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/orchestration/service/CCDAOrchestrationService.java (hunk 2 above)
Comment (ddecouteau) [reply leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/orchestration/service/CCDAOrchestrationService.java]: will correct! and re-commit

File: leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/model/card/Card.java (hunk 1 above)
Comment (mojitoj) [reply leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/model/card/Card.java:2]: I see. I wonder if we can move the java client library back to 1.7. I don't think there is anything there that depends on newer java features.

File: leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/card/ConsentConsultCardClient.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  private static final Header CDS_CLIENT_HEADER_ACCEPTS = new BasicHeader(HttpHeaders.ACCEPT, "application/json");
+
+
+  public ConsentConsultCardClient(String host) {
+    this.host = host;
+  }
+
+  public PatientConsentConsultHookResponse getConsentDecision(String request) throws IOException {
+
+//    URL url = new URL(host + endpoint);
```
Comment (mojitoj) [inline leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/card/ConsentConsultCardClient.java]: maybe we can remove these commented-out sections to clean up.

File: leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/card/ConsentConsultCardClient.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        LOGGER.log(Level.INFO, String.format("CDS Response: ", sb.toString()));
+
+        result = sb.toString();        
+        
+        System.out.println("******* CDS DECISION RESULT *********");
+        System.out.println(result);
+
+        decision = new ObjectMapper().readValue(result, PatientConsentConsultHookResponse.class);
+
+//        OutputStream os = conn.getOutputStream();
```
Comment (mojitoj) [inline leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/card/ConsentConsultCardClient.java]: ditto

Comment (ddecouteau) [reply leap-ces-ccda-orchestration-embedded/src/main/java/gov/hhs/onc/leap/ces/common/clients/card/ConsentConsultCardClient.java]: see latest commit

**Pull request merged**